src/main/java/com/netflix/imflibrary/st0429_9/AssetMap.java

```diff
@@ -106,57 +125,72 @@ public AssetMap(ResourceByteRangeProvider resourceByteRangeProvider, @Nonnull IM
 
         int numErrors = (imfErrorLogger != null) ? imfErrorLogger.getNumberOfErrors() : 0;
         validateAssetMapSchema(resourceByteRangeProvider, imfErrorLogger);
-
-        try(InputStream inputStream = resourceByteRangeProvider.getByteRangeAsStream(0, resourceByteRangeProvider.getResourceSize()-1);
-            InputStream assetMap_schema_is = AssetMap.class.getResourceAsStream(AssetMap.assetMap_schema_path);)
-        {
-            SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI );
-            StreamSource schemaSource = new StreamSource(assetMap_schema_is);
-            Schema schema = schemaFactory.newSchema(schemaSource);
-
-            ValidationEventHandlerImpl validationEventHandlerImpl = new ValidationEventHandlerImpl(true);
-            JAXBContext jaxbContext = JAXBContext.newInstance("org.smpte_ra.schemas.st0429_9_2007.AM");
-            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-            unmarshaller.setEventHandler(validationEventHandlerImpl);
-            unmarshaller.setSchema(schema);
-
-            JAXBElement<AssetMapType> assetMapTypeJAXBElement = (JAXBElement)unmarshaller.unmarshal(inputStream);
-            if(validationEventHandlerImpl.hasErrors())
-            {
-                List<ValidationEventHandlerImpl.ValidationErrorObject> errors = validationEventHandlerImpl.getErrors();
-                for(ValidationEventHandlerImpl.ValidationErrorObject error : errors){
-                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_AM_ERROR, error.getValidationEventSeverity(), error.getErrorMessage());
+        JAXBElement assetMapTypeJAXBElement = null;
+        AssetMapSchema assetMapSchema = null;
+
+        for(int i=0; i<supportedAssetMapSchemas.size(); i++) {
```

Comment:
Can any inference from "validateAssetMapSchema()" be used to avoid looping over supported schemas again?